uint8 trans_id
uint8 trans_shoes
uint8 trans_location
